contracts/zero-ex/contracts/src/storage/LibOwnableStorage.sol

```diff
@@ -0,0 +1,41 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+
+/// @dev Storage helpers for the `Ownable` feature.
+library LibOwnableStorage {
+
+    /// @dev Globally unique offset for the storage bucket.
+    bytes32 constant internal STORAGE_ID =
+        0xeef73acb590dd70cb88ccc8e9832ea7f198de2f3c87ff92d610497d647795b3c;
```

Comment:
Yeah.

Reply:
What would you guys think of a pattern like this? 
```
library LibStorage {
    enum Buckets {
        Proxy,
        Ownable,
        SimpleFunctionRegistry
    }
}

library LibProxyStorage {
    function getStorage() internal pure returns (Storage storage stor) {
        uint256 storageId = uint256(LibStorage.Buckets.Proxy);
        assembly { stor_slot := storageId }
    }
}
```

Would make it easier to verify that code is not accidentally writing to the wrong slot in review. Also tracks which bucket points to which slot when a bucket is deprecated (ie, would change `Proxy` to like `Proxy_Deprecated_v200`).

Reply:
We could move the IDs into a single file but I'm leery of using an enum because the slots would be right next to each other. So might two structs with more than one value field overlap?

Reply:
Yeah, this reverts.

```solidity
pragma solidity ^0.6.5;

contract BucketTest {
    struct Bucket1 {
        uint256 a;
        uint256 b;
    }
    
    struct Bucket2 {
        uint256 c;
    }
    
    function test() external {
        Bucket1 storage b1;
        Bucket2 storage b2;
        assembly {
            b1_slot := 1
            b2_slot := 2
        }
        b2.c = 123;
        b1.a = 456;
        b1.b = 789;
        require(b2.c == 123, 'OVERLAP_DETECTED');
    }
}
```

It would work if we hashed the enum right beforehand but that introduces overhead and these get called a lot.

Reply:
Or how about we just multiply the storage ID by a big number like `storageId * 1e6` to space them out?

Reply:
why not just `bytes32 constant internal STORAGE_ID = keccak256(Buckets.Proxy)` ?

Reply:
Ah that's right, good catch! I'm down for `storageId * 1e6`.

Reply:
Because keccak in a constant still gets run every time in 0.6 😟